ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -274,5 +331,11 @@ private void addRequiredColumn(final ColumnName columnName) {
           column.index()
       );
     }
+
+    private Boolean shouldSetInputType(final FunctionCall node, final CodeGenTypeContext context) {
```

Comment:
I don't think we should be hard coding function names in here to know to put arguments into the context. Because of the syntax rule, we're guaranteed that only Lambdas will be at the end of a function call and no more arguments will be after them. We should just be able to put any argument we encounter into the context.

Reply:
I agree hard coding is unideal, but how does knowing lambdas are at the end help us determine what arguments to put into the context? The issue for `reduce_map(map, s, (k,v,s) ...)` is that we want to store the input type of `s` but don't want to double store x in something like `transform_array(arr, x => ucase(x))`, how will we distinguish between the cases?

Reply:
Once lambda function start appearing, we should enforce that the number of args for each of the lambdas matches what we've stored so far.

So if we had a map and then another argument before lambdas, we store two input types from the map (assume order is key then value) and one input type for the other argument, we should make sure that all lambdas that occur afterward have 3 arguments so that we can map it exactly to what we've stored.

In the transform_array case, we encounter only the array before the lambda so we have one input type stored and wouldn't expect the lambda to be something like `(x,y) =>...`

Reply:
It shouldn't matter what the function name is, we should just be keeping track of the input types we've encountered while processing the function's arguments

Reply:
Mmm I see what you mean, that should work